sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -73,6 +89,17 @@ internal static DetectedLanguage ConvertToDetectedLanguage(DocumentLanguage docu
             return new DetectedLanguage(documentLanguage.DetectedLanguage, ConvertToWarnings(documentLanguage.Warnings));
         }
 
+        // CHANGED: added overload for New Swagger
+        internal static DetectedLanguage ConvertToDetectedLanguage(LanguageDetectionDocumentResult documentLanguage)
+        {
+            List<TextAnalyticsWarning> warnings = new List<TextAnalyticsWarning>();
+            foreach (var warning in documentLanguage.Warnings)
```

Comment:
If the service doesn't return `Warnings`, `document.Language.Warnings` is going to be null and the code here is going to throw.

Because this check and loop happen in every endpoint, we extracted that logic into the [ConvertToWarnings](https://github.com/Azure/azure-sdk-for-net/blob/2c9bf3756510134d789bf86e85c3ce7ab013b6d5/sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs#L67) method.
We should do the same or similar for new code too.

Reply:
added a ConvertToWarnings method